db/migrate/20200331111947_add_reporting_org_association_to_activity.rb

```diff
@@ -0,0 +1,37 @@
+class AddReportingOrgAssociationToActivity < ActiveRecord::Migration[6.0]
+  def up
+    add_reference :activities, :reporting_organisation, type: :uuid, foreign_key: {to_table: :organisations}
+
+    ActiveRecord::Base.transaction do
+      activities = Activity.where(reporting_organisation: nil)
+      service_owner = Organisation.find_by_service_owner(true)
+
+      activities.each do |activity|
+        organisation = activity.organisation
+        activity.reporting_organisation = if organisation.is_government?
+          service_owner
+        else
+          organisation
+        end
+        activity.save(validate: false)
+      end
+    end
+
+    remove_column :activities, :reporting_organisation_reference, :string
+  end
+
+  def down
+    add_column :activities, :reporting_organisation_reference, :string
+
+    ActiveRecord::Base.transaction do
+      activities = Activity.where(reporting_organisation_reference: nil)
```

Comment:
As we've just added that column I'm not sure we need this `where` query. Perhaps instead we do want to only run this code on activities that have an associated `reporting_organisation` present?

Reply:
Would `activities = Activity.where.not(reporting_organisation_id: nil)` work here?

Reply:
Yes 🙌